### ISSUE TITLE
packages.py version not respected depending on path setting

### ISSUE THREAD

**Issue by ferdonline** [spack/spack#9309]
Consider defining the default package version, using "version" in packages.yaml.
This setting is discarded when 'path' entries are also added without including a matching  version.

### Explanation

Consider the following use case (which apparently can't be met):
 - There is a system Python which shall be used if the version is the one desired (e.g.: 2.7.5)
 - By default software should build on top of a newer (e.g.: 2.7.15)

It seems that it could be specified by adding to packages.yaml the following definitions:
```
python:
    paths:
        python@2.7.5: /usr
    version: [2.7.15]
```
However this doesn't work, i.e. spack will still use python@2.7.5 by default.

Is it the expected behavior?

**Comment by healther** [comment]
What do you mean by use? 

```
spack install python
```

should default to `python@2.7.15`, on the other hand

```
spack load python
```

will try to load _everything_ that is installed and matches `python`.

I.e. if you haven't installed `python@2.7.15` it will try to load your system one (I'm not too sure what exactly is going to happen there).
If you have installed `python@2.7.15` (e.g. by doing a `spack install python`) I would expect it to error out, as you ask it to load more than one version and you will have to be more precise in what you actually want.

At least this is what I would consider expected behaviour, but having an external package in there might screw things up (I've never had to work with these)

**Comment by pramodk** [comment]
> What do you mean by use?
> spack install python
> should default to python@2.7.15

No, this still install `python@2.7.5`. This confusion arises because version preferences with external packages doesn't work as expected (known issue, discussed in #5110).

**Comment by alalazo** [comment]
These kind of preferences are the lowest priority, as explained [here](https://spack.readthedocs.io/en/latest/frequently_asked_questions.html). Externals can be considered the same priority as "reuse" (so higher), and they will actually be after #51118 Closing this issue, but let me know if anything still needs clarifications.
